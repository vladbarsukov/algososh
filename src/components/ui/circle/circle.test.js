import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';
import {Circle} from "./circle";


describe('Test CircleComponent', () => {

  it("should render and Match Snapshot with text", () => {
    const CircleComponent = renderer.create(<Circle />).toJSON();
    render(<Circle />)
    expect(CircleComponent).toMatchSnapshot();
  });

});