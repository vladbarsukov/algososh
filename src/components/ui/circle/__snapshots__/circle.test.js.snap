// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CircleComponent should render and Match Snapshot with changing state 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with component in head 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content "
      data-testid="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head string"
        data-testid="head"
      >
        test
      </div>
      <div
        className="circle   default"
        data-testid="circleWrapper"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with component in tail 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content "
      data-testid="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head string"
        data-testid="head"
      >
        test
      </div>
      <div
        className="circle   default"
        data-testid="circleWrapper"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with default state 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with head 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with index 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with letter 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      a
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with modified state 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with props: isSmall ===  true 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot with tail 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Test CircleComponent should render and Match Snapshot without letter 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleWrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
